### DISCUSSION TITLE
Can i use this without composer?

### DISCUSSION THREAD

**Discussion by codemasher** [chillerlan/php-qrcode#62] [README]
# Can i use this library without using composer?

## You can, but it's absolutely not recommended, nor supported. 

With that said, i'll leave you with this info:

- [download the .zip](https://github.com/chillerlan/php-qrcode/releases) for a version of your choice and also of any required (not dev) dependency listed in the `composer.json` for that version (you can find links to the respective repos [on packagist](https://packagist.org/packages/chillerlan/php-qrcode))
- extract the files into your library folder
- include the files manually or with whatever autoloader you are using
- good luck!

~~Maybe at some point in the future i may add a .phar release that includes all necessary files,~~ until then... just use composer. ([see below](#discussioncomment-2199209))

Oh and please do not come over to complain about a "vendor monopoly" (we had that before) and remember that you are on GitHub that is owned by Microsoft...

**Comment by morrow95** [comment]
@codemasher +1 for a phar release as I prefer not to install or use composer either.

**Reply by codemasher** [reply]
I don't think a phar release makes much sense for a single library that's supposed to be part of another release. If you think otherwise, feel free to submit a pull request for packaging,uploading and attaching a .phar to release during the build process. If you don't want to use composer because of *\<insert whatever misled belief\>*, i can't help you but to follow the steps i mentioned above.

**Reply by accountnujen** [reply]
@codemasher
Your app has grown into a huge combine where you can create QR codes in PDF, both with and without a logo, and this way and that. But what if I don't need all that cool functionality? What if I just need to do this: `<img src="<?= $qrpng ?>" >`. What's the minimum set of files I need to plug in to just generate PNGs (or SVGs, whichever is easier)?

I don't want to drag in all the stuff you won't use, so I want to have instructions [like PHPMailer](https://github.com/PHPMailer/PHPMailer#:~:text=Alternatively%2C%20if%20you%27re%20not%20using%20Composer):

**Reply by codemasher** [reply]
Hi. `composer install --no-dev` already installs the absolute minimum required set of files - if that is still too much for you, i can't help you. Feel free to try and delete from that set of files what you feel you don't need.
